app/scripts/services/images.js

```diff
@@ -0,0 +1,216 @@
+'use strict';
+
+angular.module("openshiftConsole")
+  .factory("ImagesService", function($filter, ApplicationGenerator, DataService) {
+    var findImage = function(name, projectContext) {
+      var importImage = {
+        kind: "ImageStreamImport",
+        apiVersion: "v1",
+        metadata: {
+          name: "newapp",
+          namespace: projectContext.namespace
+        },
+        spec: {
+          import: false,
+          images: [{
+            from: {
+              kind: "DockerImage",
+              name: name
+            }
+          }]
+        },
+        status: {}
+      };
+
+      return DataService.create('imagestreamimports',
+                                null,
+                                importImage,
+                                projectContext).then(function(response) {
+        return {
+          name: _.get(response, 'spec.images[0].from.name'),
+          image: _.get(response, 'status.images[0].image'),
+          tag: _.get(response, 'status.images[0].tag'),
```

Comment:
same question here, should we be handling id based pull specs, or does imagestreamimport not support that?